crates/tests/mqtt_tests/src/test_mqtt_server.rs

```diff
@@ -86,69 +89,76 @@ fn spawn_broker(port: u16) {
     });
 
     std::thread::spawn(move || {
-        let mut rx = tx.connect(200).unwrap();
         tx.subscribe("#").unwrap();
 
         loop {
-            if let Some(message) = rx.recv().unwrap() {
-                for chunk in message.payload.into_iter() {
-                    let mut bytes: Vec<u8> = vec![];
-                    for byte in chunk.into_iter() {
-                        bytes.push(byte);
-                    }
-                    let payload = match std::str::from_utf8(bytes.as_ref()) {
+            let notification = match rx.recv().unwrap() {
+                Some(v) => v,
+                None => continue,
+            };
+            match notification {
+                Notification::Forward(forward) => {
+                    let payload = match std::str::from_utf8(&forward.publish.payload) {
                         Ok(payload) => format!("{:.110}", payload),
-                        Err(_) => format!("Non uft8 ({} bytes)", bytes.len()),
+                        Err(_) => format!("Non uft8 ({} bytes)", forward.publish.payload.len()),
                     };
                     eprintln!(
-                        "MQTT-TEST MSG: topic = {}, payload = {:?}",
-                        message.topic, payload
+                        "MQTT-TEST MSG: topic = {:?}, payload = {:?}",
+                        forward.publish.topic, payload
                     );
                 }
+
+                v => {
```

Comment:
What are these `Notification` variants and is it worth printing those? If we're gonna keep it, just add a comment mentioning what these `v`s are.

Reply:
Sure, I will add the comment there. These notifications are from the broker for the notifications like `Forward, acks,  etc.

Reply:
Not very useful though: e.g. `DeviceAck(SubAck(SubAck { pkid: 0, return_codes: [QoS0] }))`

```
---- tests::tests::ensure_that_last_will_message_is_delivered stdout ----
MQTT-TEST INFO: start test MQTT broker (port = 55555)
DeviceAck(SubAck(SubAck { pkid: 0, return_codes: [QoS0] }))
MQTT-TEST MSG: topic = b"test/lwp", payload = "hello 1"
MQTT-TEST MSG: topic = b"test/lwp", payload = "hello 2"
MQTT-TEST MSG: topic = b"test/lwp", payload = "hello 3"
```

What would be great is to trace new connection as well as disconnection. I would ignore all other `rumqttd` internal messages.

Reply:
Sure, I will not print anyother notifications.